core/src/main/java/com/sequenceiq/cloudbreak/service/smartsense/SmartSenseSubscriptionService.java

```diff
@@ -104,27 +108,55 @@ public SmartSenseSubscription findOneById(Long id) {
         return subscriptions.hasNext() ? Optional.of(subscriptions.next()) : Optional.empty();
     }
 
-    @PostAuthorize("hasPermission(returnObject,'read')")
-    public SmartSenseSubscription getDefaultForUser(IdentityUser cbUser) {
-        SmartSenseSubscription subscription;
-        subscription = repository.findByAccountAndOwner(cbUser.getAccount(), cbUser.getUserId());
+    public SmartSenseSubscription getDefaultForUser(@Nonnull IdentityUser cbUser) {
+        requireNonNull(cbUser);
+        SmartSenseSubscription subscription = repository.findByAccountAndOwner(cbUser.getAccount(), cbUser.getUserId());
+        upgradeDefaultSmartSenseSubscription(subscription);
```

Comment:
I would simplify this logic with an easily readable if statement and try to create the subscription when no entity could be found in the database, only try to upgrade the subscription when any exists.